addon/ext.js

```diff
@@ -47,8 +47,15 @@ Store.reopen({
   createFragment: function(modelName, props) {
     Ember.assert("The '" + modelName + "' model must be a subclass of MF.Fragment", this.isFragment(modelName));
 
-    var type = this.modelFor(modelName);
-    var internalModel = new InternalModel(type, null, this, getOwner(this).container);
+    var internalModel;
+    if (InternalModelPrototype.hasOwnProperty('modelClass')) {
```

Comment:
Oh, I didn't know that! Thanks for catching this. Yea, I like your idea of just moving it outside the function. I'll make it so.